Sprint_2

mrAlexCorpQA opened the pull request:
Спринт 2. Финальный проект: Обязательное задание.

Commit:
Спринт 2. Финальный проект: Обязательное задание.

egorvas commented on .idea/.gitignore:
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
```
Рекомендую добавлять папку .idea в .gitignore всего репозитория. По сути файлы в папке .idea это твои локальный файлы и не нужны для работоспособности проекта, но при этом могут приватные данные.

mrAlexCorpQA replied:
Исправлено. Параметры файла .gitignore изменены

egorvas commented on src/test/java/TestCat.java:
```diff
@@ -0,0 +1,56 @@
+
+    //Проверяется исключение метода getFood класса Cat (замокирована зависимость)
+    @Test
+    public void testCatGetFoodException() {
```
Я не вижу смысла в этом тесте если честно. Он полностью идентичен предыдущему, просто ты ответ массивом заменил на эксепшен. В моем понимании это абсолютно бесполезно.

mrAlexCorpQA replied:
Исправлено. Тест удален

egorvas commented on src/test/java/TestFelineParametrized.java:
```diff
@@ -0,0 +1,39 @@
+            };
+        }
+
+        private int a;
```
Всегда давай переменным понятные имена

mrAlexCorpQA replied:
Исправлено. Переменная переименована

egorvas commented on src/test/java/TestLionNoneParametrized.java:
```diff
@@ -0,0 +1,68 @@
+
+    //Проверяется исключение метода getFood класса Lion (замокирована зависимость)
+    @Test
+    public void testLionGetFoodException() {
```
Опять бесполезный тест на мой взгляд

mrAlexCorpQA replied:
Исправлено. Тест удален

egorvas commented on src/test/java/TestLionNoneParametrized.java:
```diff
@@ -0,0 +1,68 @@
+
+    //Проверяется передаваемое исключение метода doesHaveMane класса Lion
+    @Test
+    public void testLionDoesHaveManeException() throws Exception {
```
Если ты хочешь проверять через try/catch, то тебе обязательно нужно проверять что твой catch состоялся. Иначе если эксепшена не будет, то твой тест вместо того чтобы упасть - пройдет, так как попросту не будет ассертов.

mrAlexCorpQA replied:
Исправлено. Тест изменен

egorvas commented on target/maven-archiver/pom.properties:
```diff
@@ -0,0 +1,5 @@
+#Generated by Maven
```
Рекомендую подобные файлы также убирать в .gitignore по аналогии с .idea
Они не нужны для работы проекта. Вообще как правило всю папку target убирают в gitignore, но так как по условиям задачи нужно было приложить отчет jacoco, то можно оставить как есть или убрать в gitignore всю папку target/ а target/site/ добавить в исключения.

mrAlexCorpQA replied:
Исправлено. Параметры файла .gitignore изменены

Commit:
Спринт 2. Финальный проект: Обязательное задание (правка замечаний).
Исправлено: убраны тесты на проверку исключения метода GetFood классов Lion и Cat

Commit:
Спринт 2. Финальный проект: Обязательное задание (правка замечаний).
Исправлено:
 1. Убраны тесты на проверку исключения метода GetFood классов Lion и Cat
 2. Переименована переменная в параметризованном тесте метода getKittens класса Feline с "а" на "valueForTest"

Commit:
Спринт 2. Финальный проект: Обязательное задание (правка замечаний).
Исправлено:
 1. Убраны тесты на проверку исключения метода GetFood классов Lion и Cat;
 2. Переименована переменная в параметризованном тесте метода getKittens класса Feline с "а" на "valueForTest";
 3. Изменен тест  метода doesHaveMane класса Lion согласно замечаниям;

Commit:
Спринт 2. Финальный проект: Обязательное задание (правка замечаний).
Исправлено:
 1. Убраны тесты на проверку исключения метода GetFood классов Lion и Cat;
 2. Переименована переменная в параметризованном тесте метода getKittens класса Feline с "а" на "valueForTest";
 3. Изменен тест  метода doesHaveMane класса Lion согласно замечаниям;
 4. Изменены параметры gitignore - скрыта ненужная информация.
